### ISSUE TITLE
ArrayTypeStyle false positives in nested generics

### ISSUE THREAD

**Issue by jdussouillez** [checkstyle/checkstyle#10215]
Rule : [ArrayTypeStyle](https://checkstyle.sourceforge.io/config_misc.html#ArrayTypeStyle)

The `ArrayTypeStyle` seems to fail when checking the declaration inside nested generics. It works as expected with the 8.43 but fails with the [8.44](https://checkstyle.sourceforge.io/releasenotes.html#Release_8.44).

```java
// Test.java
public class Test {

    protected Pair<Integer, Object>[] values1; // OK
    protected Pair<Integer, Pair<String, Object>[]>[] values2; // OK
    protected Pair<Integer, Pair<String, Pair<String, Object>>[]>[] values3a; // KO!
    protected Pair<
        Integer,
        Pair<
            String,
            Pair<String, Object>
        >[]
    >[] values3b; // OK, but in fact it's the same as "values3a" with another format

    protected static class Pair<L, R> {
        protected L key;
        protected R value;
    }
}
```

```xml
<!-- checkstyle.xml -->
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="error"/>

    <module name="SuppressWithPlainTextCommentFilter"/>
    <property name="fileExtensions" value="java, properties, xml"/>
    <module name="TreeWalker">
        <module name="ArrayTypeStyle"/>
    </module>
</module>
```

```bash
$ java -jar checkstyle-8.44-all.jar -c checkstyle.xml Test.java
```
```
Starting audit...
[ERROR] /home/junior/Downloads/Test.java:5:66: Array brackets at illegal position. [ArrayTypeStyle]
Audit done.
Checkstyle ends with 1 errors
```

**Comment by strkkk** [comment]
@nmancus1 isn't this regression from recent types changes in AST, e.g. https://github.com/checkstyle/checkstyle/issues/10100?

**Comment by nmancus1** [comment]
Sure is. Please approve, and I will fix this right away.

**Comment by romani** [comment]
Fix is merged
